transcendence-app/src/game/infrastructure/db/postgres/game.postgres.repository.ts

```diff
@@ -71,4 +76,34 @@ export class GamePostgresRepository
     });
     return gamesData ? gamesData.map((game) => new this.ctor(game)) : null;
   }
+
+  async getPaginatedUserGamesWithUsers(
+    username: string,
+    paginationDto: Required<PaginationWithSearchQueryDto>,
+  ): Promise<GameWithUsers[] | null> {
+    const { limit, offset, sort, search } = paginationDto;
+    const orderBy =
+      sort === BooleanString.True ? gameKeys.CREATED_AT : gameKeys.ID;
+    const gamesData = await makeQuery<GameWithUsersData>(this.pool, {
+      text: `SELECT g.*,
+      ${renameColumnsWithPrefix('p1', Object.values(userKeys), 'playerOne')},
+      ${renameColumnsWithPrefix('p2', Object.values(userKeys), 'playerTwo')}
+      FROM ${this.table} g
+      JOIN ${table.USERS} AS p1 ON g.${gameKeys.PLAYERONEUSERNAME} = p1.${
+        userKeys.USERNAME
+      }
+      JOIN ${table.USERS} AS p2 ON g.${gameKeys.PLAYERTWOUSERNAME} = p2.${
+        userKeys.USERNAME
+      }
+      WHERE (g.${gameKeys.PLAYERONEUSERNAME} = $4
+        AND g.${gameKeys.PLAYERTWOUSERNAME} ILIKE $1)
```

Comment:
why do we want to make this case insensitive?

Reply:
I don't mind, we can change it to LIKE $1 if you want. But we should be consistent with the other search queries

Reply:
I would say when we're searching user input, we need the case insensitiveness, and the '%search%' percents completions. When we're using our own database data, we should use an exact match, because it will work, it's unambiguous, and it's lighter.

Reply:
opinion: Here we retrieve a user's games (by their username) and filter the opponents by the search input.

Let's say we have a user `u` with the following games:

> u vs a
u vs Alice
allison vs u
Arthur vs u
alien vs u


If we want to filter the oponent by `a`:

With a case-sensitive search, we get the following results:

> u vs **a**
**a**llison vs u
**a**lien vs u

With a case-insensitive search, we get the following results:

> u vs a
u vs **A**lice
**a**llison vs u
**A**rthur vs u
**a**lien vs u

If we want to filter the oponent by `al`:

With a case-sensitive search, we get the following results:
> **al**lison vs u
**al**ien vs u

With a case-insensitive search, we get the following results:

> u vs **Al**ice
**al**lison vs u
**al**ien vs u

The case-insensitive provides more matches, and it's more consistent with our other endpoints (users, chatrooms and chatroom members)

Reply:
I'm opening a new issue about case-sensitive searches just in case we want to change it. By now, I going to leave it as it is